src/tiled/newtemplatedialog.cpp

```diff
@@ -27,13 +27,20 @@
 using namespace Tiled;
 using namespace Tiled::Internal;
 
-NewTemplateDialog::NewTemplateDialog(QList<QString> groupNames, QWidget *parent) :
+NewTemplateDialog::NewTemplateDialog(const QList<QString> &groupNames, const QString &objectName, QWidget *parent) :
     QDialog(parent),
     mUi(new Ui::NewTemplateDialog)
 {
     mUi->setupUi(this);
+
+    mObjectName = objectName;
+    mUi->templateNameCheckBox->setEnabled(!objectName.isEmpty());
```

Comment:
Why did you choose to have a checkbox? I think it will be enough to auto-fill the field with the name of the object.